### ISSUE TITLE
Standard rules (MD041, MD022, MD007) false-positive on YAML frontmatter in 0.14.4

### ISSUE THREAD

**Issue by juherr** [joshrotenberg/mdbook-lint#406]
### Version

mdbook-lint 0.14.4 (CLI, `lint` subcommand)

### Summary

The standard Markdown rules don't honour a document's YAML frontmatter: any file that starts with a `---` frontmatter block gets false positives from MD041 ("first line should be a top-level heading") and MD022 ("heading should be surrounded by blank lines"), and MD007 when the frontmatter contains a block sequence. The `---` delimiters and the YAML keys are linted as if they were Markdown headings / lists.

This happens even though:

- comrak is configured to recognise frontmatter (`document.rs`: `options.extension.front_matter_delimiter = Some("---")`), and
- MD041 documents a `front_matter_title` option for exactly this case.

So the parser recognises the frontmatter node, but the standard rules don't consult it.

### Reproduction

`doc.md`:

```markdown
---
title: My Document
status: accepted
---

# My Document

A short paragraph of body text.
```

Run:

```console
$ mdbook-lint lint --enable MD041,MD022,MD007 doc.md
warning[MD041]: First line in file should be a top level heading
  --> doc.md:1:1
warning[MD022]: Heading should be followed by a blank line
  --> doc.md:1:1
Found: 2 warning(s)
```

The same body **without** the frontmatter block is clean (`No issues found`).

### Expected

With frontmatter present and recognised, MD041 should treat the document as having a title (it has a `title:` field), and MD022 should not see the `---` fence / YAML keys as headings. → no warnings.

### Actual

MD041 and MD022 fire on line 1 (the opening `---`).

### `front_matter_title` does not help

Configuring it explicitly still leaves MD041 firing:

```toml
# cfg.toml
[MD041]
front_matter_title = "^\\s*title\\s*[:=]"
```

```console
$ mdbook-lint lint -c cfg.toml --enable MD041 doc.md
warning[MD041]: First line in file should be a top level heading
  --> doc.md:1:1
Found: 1 warning(s)
```

### MD007 (block-sequence frontmatter)

A MADR-style frontmatter with a block sequence also trips MD007:

```markdown
---
status: accepted
decision-makers:
  - Alice
---

# Title

Body.
```

`mdbook-lint lint --enable MD007 …` reports the `  - Alice` line as a mis-indented list.

### Notes

- Real-world impact: on docs with frontmatter + actual content the misparsed block context cascades into further false positives (MD032, MD004).
- The `adr` provider added in 0.14.4 parses MADR frontmatter correctly, so the capability exists in the codebase — it's the standard ruleset that doesn't use it.

### Affected rules (confirmed)

MD041, MD022, MD007 directly; MD032 / MD004 via cascade on real content.

### Environment

- mdbook-lint 0.14.4
- macOS (aarch64-apple-darwin)
